ui/src/selection-panel/selection-panel-geometry-section.tsx

```diff
@@ -95,7 +95,7 @@ const SelectionPanelGeometrySection: React.FC<GeometryPlansPanelProps> = ({
             setPlanHeadersInView(page.items);
             setPlanHeaderCount(page.totalCount);
         });
-    }, [viewport.area, changeTimes.geometryPlan, selectedTrackNumbers]);
+    }, [viewport.area, changeTimes.geometryPlan, selectedTrackNumbers.sort().join('')]);
```

Comment:
Vaikka sillä ei sinällään niin väliä olekaan (ei pitäisi olla koskaan käyttäjän näkemä arvo), niin jos nämä joinataan, niin ne voisi joinata toisiinsa alaviivalla. Also, olikos tässä jotain minkä takia tuo aiempi array ei toimisi tässä?

Reply:
pitäis ehkä muutenkin miettiä miten me passaillaan propseja koska nyt pienikin muutos redux-storessa aiheuttaa redrawin kaikkialla (koska state olio on "uusi")

Reply:
Tuon selected-joukon kannattaisi varmaan olla sortattu jo redux-storessa ja siellä voisi myös varmistaa että ei muuteta arraytä jos sisältö on jo sama. Silloin kaikissa käyttöpaikoissa ei tarvitse tehdä tällaista kikkailua vaan voi luottaa että array muuttuu vain jos se muuttuu

Reply:
selectedTrackNumbers generoidaan lennosta --> minkä tahansa  yläkomponentin propsun muutos laukaisee kans muutoksen täällä.

Reply:
ai se ei tulekaan storesta? meh, no eipä sitten